pandas/tests/io/formats/test_to_csv.py

```diff
@@ -644,6 +648,27 @@ def test_to_csv_encoding_binary_handle(self, mode):
                 handle.seek(0)
                 assert handle.read().startswith(b'\xef\xbb\xbf""')
 
+    @pytest.mark.parametrize(
+        "df,csv_name",
+        [
+            (
+                DataFrame({"a": [1, 2, 3, 4, 5], "b": [4, 5, 6, 7, 8]}),
+                "test_to_csv_zipped_content_name.csv",
+            )
+        ],
+    )
+    def test_to_csv_content_name_in_zipped_file(self, df, csv_name):
+        suffix_zip_name = csv_name + ".zip"
+        with tm.ensure_clean(suffix_zip_name) as pth:
+            # ensure_clean will add random str before suffix_zip_name,
+            # need Path.stem to get real file name
+            df.to_csv(pth)
+            pp = Path(pth)
+            with ZipFile(pth) as zf:
+                result = zf.filelist[0].filename
+                expected = pp.stem
+                assert result == expected
```

Comment:
Personal preference would be to simply `assert zf.filelist[0].filename == Path(path).stem` instead of this incremental approach. I think it's way quicker to read and understand. Also, better to avoid using `pp`, `pth` and other cryptic names, descriptive names even if a bit longer make everyone's life easier IMHO.